cpp/include/cudf/binaryop.hpp

```diff
@@ -72,7 +72,15 @@ enum class binary_operator : int32_t {
                          ///< operand when one is null; or invalid when both are null
   GENERIC_BINARY,        ///< generic binary operator to be generated with input
                          ///< ptx code
-  INVALID_BINARY         ///< invalid operation
+  NULL_LOGICAL_AND,      ///< operator && with Spark rules: NULL_LOGICAL_AND(null, null) is null,
+                         ///< NULL_LOGICAL_AND(null, true) is
+  ///< null, NULL_LOGICAL_AND(null, false) is false, and NULL_LOGICAL_AND(valid,
```

Comment:
indentation seems a little off here and in the next line.

Reply:
clang format is doing it. I have made a few changes to try and reduce it some, but it is still the format tool that is making it happen.

Reply:
FYI, clang-format can be disabled over a stretch of code with `// clang-format off`, followed by `// clang-format on`. Examples: https://github.com/rapidsai/cudf/search?q=%2F%2F+clang-format+off